evalml/tests/data_checks_tests/test_datetime_format_data_check.py

```diff
@@ -0,0 +1,130 @@
+import pandas as pd
+import pytest
+
+from evalml.data_checks import (
+    DataCheckError,
+    DataCheckMessageCode,
+    DateTimeFormatDataCheck,
+)
+
+datetime_format_check_name = DateTimeFormatDataCheck.name
+
+
+@pytest.mark.parametrize("input_type", ["pd", "ww"])
+@pytest.mark.parametrize(
+    "uneven,type_errors", [(True, False), (False, True), (False, False)]
+)
+@pytest.mark.parametrize("datetime_loc", ["datetime_feature", "X_index", "y_index"])
+def test_datetime_format_data_check_typeerror_uneven_intervals(
+    uneven, input_type, type_errors, datetime_loc
+):
+    X, y = pd.DataFrame({"features": range(30)}), pd.Series(range(30))
+
+    if type_errors:
+        dates = range(30)
+    else:
+        dates = pd.date_range("2021-01-01", periods=30)
+
+    if uneven:
+        dates = list(dates)
+        dates[-1] = "2021-01-31"
```

Comment:
Changes `2021-01-30` to `2021-01-31`, skipping a date and triggering the error

Reply:
Could be good to mention this as a comment :)